### PR TITLE
[stable8.1] Fix enc wrapper without encryption stable8.1

### PR DESCRIPTION
**Pull request by DeepDiver1975**
backport of #17840

### CODE REVIEW

File: apps/files/appinfo/update.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ */
+
+/**
+ * @param \OCP\IDBConnection $conn
+ */
+function owncloud_reset_encrypted_flag(\OCP\IDBConnection $conn) {
+	$conn->executeUpdate('UPDATE `*PREFIX*filecache` SET `encrypted` = 0 WHERE `encrypted` = 1');
+}
+
+// Current version of ownCloud before the update is 8.1.0 or 8.2.0.(0-2)
```
Comment (PVince81) [inline apps/files/appinfo/update.php]: Just tested, the migration didn't work. It still left the "encrypted" flags to 1.

Do these version numbers need changing ? @schiesbn

Comment (DeepDiver1975) [inline apps/files/appinfo/update.php]: What was the version you did com from @PVince81 ?

Comment (PVince81) [inline apps/files/appinfo/update.php]: The one from the current stable8.1 which is 8.1.1.0.

Comment (PVince81) [inline apps/files/appinfo/update.php]: I also wondered why the version compare utility function wasn't used here.

Comment (DeepDiver1975) [inline apps/files/appinfo/update.php]: > I also wondered why the version compare utility function wasn't used here.

@schiesbn any idea?

Comment (schiesbn) [inline apps/files/appinfo/update.php]: I know that @nickvergessen had some reasons, but I don't remember.

File: apps/files/appinfo/update.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+if ($installedVersion === '1.1.9' && (
+		// 8.1.0.x
+		(((int) $ocVersion[0]) === 8 && ((int) $ocVersion[1]) === 1 && ((int) $ocVersion[2]) === 0)
+		||
+		// < 8.2.0.3
+		(((int) $ocVersion[0]) === 8 && ((int) $ocVersion[1]) === 2 && ((int) $ocVersion[2]) === 0 && ((int) $ocVersion[3]) < 3)
```
Comment (PVince81) [inline apps/files/appinfo/update.php]: Change this to " < 8.1.1.1" ?

**Comment by schiesbn** [comment]
I just discussed the issue with @PVince81, my last commit should fix it. @PVince81 can you give it another try? Thanks!

**Comment by PVince81** [comment]
@owncloud-bot retest this please

**Comment by PVince81** [comment]
Migration works now from 8.1.1.0 and the flag isn't set any more when on this branch :+1:

**Comment by PVince81** [comment]
Jenkins PR: https://github.com/owncloud/core/pull/17913

**Comment by PVince81** [comment]
Fixes https://github.com/owncloud/core/issues/17904

**Comment by PVince81** [comment]
Come on Jenkins, everybody is waiting for you!

**Comment by PVince81** [comment]
@DeepDiver1975 @schiesbn could you review each other's code so we get a second thumbs up?

**Comment by schiesbn** [comment]
:+1:

**Comment by PVince81** [comment]
@cmonteroluque the status is "we're waiting for Jenkins". So if it shows green, feel free to merge.

**Comment by DeepDiver1975** [comment]
2 of 3 8.1 instances could not be migrated - both blocked instances share the fact that encryption is enabled.

from the log:

````

{"reqId":"No8qjIlPGyrrIuoE7Qc8","remoteAddr":"178.25.225.40","app":"index","message":"Exception: {\"Exception\":\"OC\\\\NeedsUpdateException\",\"Message\":\"\",\"Code\":0,\"Trace\":\"#0 \\\/srv\\\/www\\\/owncloud8.1\\\/lib\\\/private\\\/app.php(99): OC_App::loadApp('files')\\n#1 \\\/srv\\\/www\\\/owncloud8.1\\\/lib\\\/private\\\/util.php(127): OC_App::loadApps(Array)\\n#2 \\\/srv\\\/www\\\/owncloud8.1\\\/lib\\\/base.php(872): OC_Util::setupFS()\\n#3 \\\/srv\\\/www\\\/owncloud8.1\\\/index.php(40): OC::handleRequest()\\n#4 {main}\",\"File\":\"\\\/srv\\\/www\\\/owncloud8.1\\\/lib\\\/private\\\/app.php\",\"Line\":118}","level":4,"time":"2015-07-27T19:28:02+00:00"}

**Comment by DeepDiver1975** [comment]
I'll try to dig deeper tomorrow ...

File: apps/files/appinfo/update.php (hunk 1 above)
Comment (nickvergessen) [inline apps/files/appinfo/update.php]: yeah version comparison had to be changed like you did afterwards

**Pull request merged**